src/setup_payload/QRCodeSetupPayloadParser.cpp

```diff
@@ -373,4 +378,26 @@ CHIP_ERROR QRCodeSetupPayloadParser::populatePayload(SetupPayload & outPayload)
     return populateTLV(outPayload, buf, indexToReadFrom);
 }
 
+std::variant<CHIP_ERROR, std::vector<SetupPayload>> QRCodeSetupPayloadParser::Parse()
+{
+    std::string payload = ExtractPayload(mBase38Representation);
+    VerifyOrReturnError(payload.length() != 0, CHIP_ERROR_INVALID_ARGUMENT);
+
+    auto chunkCount = std::count(payload.begin(), payload.end(), '*') + 1;
+
+    std::vector<SetupPayload> retval;
+    retval.reserve(static_cast<decltype(retval)::size_type>(chunkCount));
+
+    std::stringstream chunkStream(payload);
+    std::string chunk;
+
+    while (std::getline(chunkStream, chunk, '*'))
+    {
+        auto & nextItem = retval.emplace_back();
+        ReturnErrorOnFailure(populatePayloadFromBase38Data(chunk, nextItem));
+    }
+
```

Comment:
This could be done by making a new string split utility that doens't require strinstream. Could be a follow-up, this is fine for now. Prefer not using std::stringstream as it includes exceptions whereas most algorithms do not have them in the first place (i.e. std::find-based splitting).

Reply:
Realistically, std::vector appending includes exceptions for the moment, so....

But yes, we could modify this as a followup.  https://github.com/project-chip/connectedhomeip/issues/38730 filed.